### PR TITLE
feat: add `file-length-limit` rule

### PR DESCRIPTION
**Pull request by alexandear**
This PR adds a new rule `file-length-limit`.
The rule is inspired on the eslint rule [max-length](https://eslint.org/docs/latest/rules/max-lines).

This rule has options:

- `max` (default `0`) enforces a maximum number of lines in a file, `0` means the rule is disabled;
- `skipBlankLines` (default `false`) if `true` ignore lines made up purely of whitespace;
- `skipComments` (default `false`) if `true` ignore lines containing just comments.

### CODE REVIEW

File: RULES_DESCRIPTIONS.md [hunk 1]
```diff
@@ -501,6 +502,23 @@ Example:
...
 
+## file-length-limit
+
+_Description_: This rule enforces a maximum number of lines per file, in order to aid in maintainability and reduce complexity.
+
+_Configuration_:
+
+* `max` (int) a maximum number of lines in a file (default 1000)
+* `skipComments` (bool) ignore lines containing just comments
+* `skipBlankLines` (bool) ignore lines made up purely of whitespace
```
Comment (ccoVeille) [inline RULES_DESCRIPTIONS.md:513]: Please explain the default values for these two

File: rule/file-length-limit.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+func (*FileLengthLimitRule) Name() string {
+	return "file-length-limit"
+}
+
+func countCommentLines(comments []*ast.CommentGroup) int {
+	count := 0
+	for _, cg := range comments {
+		for _, comment := range cg.List {
+			switch comment.Text[1] {
```
Comment (ccoVeille) [inline rule/file-length-limit.go:116]: I'm a bit afraid the slice could not contain the element 1, so it may panic

**Comment by ccoVeille** [comment]
For record, archives and context

The following linter made by @DanilXO has been merged into [golangci-lint](https://github.com/golangci/golangci-lint/pull/5081)

https://github.com/DanilXO/filen


It also was discussed here
https://github.com/golangci/golangci-lint/discussions/2881

**Comment by DanilXO** [comment]
SkipBlankLines is an interesting addition. I'll add it to my linter too. I was thinking about revive as a place to add this, but I liked the idea of ​​golangci-lint with little disabling linters better.

File: rule/file-length-limit.go (hunk 2 above)
Comment (alexandear) [reply rule/file-length-limit.go:119]: Added the check for `len(comment.Text)`

File: RULES_DESCRIPTIONS.md (hunk 1 above)
Comment (alexandear) [reply RULES_DESCRIPTIONS.md]: Extended slightly description

File: rule/file-length-limit.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+			},
+			Failure: fmt.Sprintf("file length is %d lines, which exceeds the limit of %d", lines, r.max),
+		},
+	}
+}
+
+func (r *FileLengthLimitRule) configure(arguments lint.Arguments) {
+	r.Lock()
+	defer r.Unlock()
+
```
Comment (chavacava) [inline rule/file-length-limit.go:74]: Here we can check if `r.max` is already set, if it is the case then we can exit.

File: rule/file-length-limit.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	argKV, ok := arguments[0].(map[string]any)
+	if !ok {
+		panic(fmt.Sprintf(`invalid argument to the "file-length-limit" rule. Expecting a k,v map, got %T`, arguments[0]))
+	}
+	for k, v := range argKV {
+		switch k {
+		case "max":
+			maxLines, ok := v.(int64)
+			if !ok || maxLines < 1 {
+				panic(fmt.Sprintf(`invalid configuration value for max lines in "file-length-limit" rule; need int64 but got %T`, arguments[0]))
```
Comment (chavacava) [inline rule/file-length-limit.go:89]:
```suggestion
				panic(fmt.Sprintf(`invalid configuration value for max lines in "file-length-limit" rule; need positive int64 but got %T`, arguments[0]))
```

**Review by chavacava — COMMENTED** [review]
Thanks @alexandear. I've left some comments. Please also add documentation to the Name method to remove the lint warn.

**Comment by chavacava** [comment]
IMO the default max must be `-1` meaning the rule is not activated. That way we will not pollute the results of those who activate all rules.

**Comment by alexandear** [comment]
> IMO the default max must be `-1` meaning the rule is not activated. That way we will not pollute the results of those who activate all rules.

@mgechev Changed.

But I choose `0` in `max` to disable the rule. This is for consistency with the `function-length` rule.

**Pull request merged**